packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -13,21 +13,25 @@ typedef constructor<T> = T Function(
 /// The [SerializableEntity] is the base class for all serializable objects in
 /// Serverpod, except primitives.
 abstract mixin class SerializableEntity {
-  /// Returns a serialized JSON structure of the model, ready to be sent
-  /// through the API. This does not include fields that are marked as
-  /// database only.
-  dynamic toJson();
-
   /// Returns a serialized JSON structure of the model which also includes
   /// fields used by the database.
-  dynamic allToJson() => toJson();
+  dynamic toJson();
 
   @override
   String toString() {
     return SerializationManager.encode(this);
   }
 }
 
+/// The [ConstraintSerialization] defines a toJson method which makes it
+/// possible to limit what fields are serialized
+abstract interface class ConstraintSerialization {
+  /// Returns a serialized JSON structure of the model, ready to be sent
+  /// through the API. This does not include fields that are marked as
+  /// database only.
+  dynamic toConstraintJson();
```

Comment:
Suggestion: Use `Map<String, dynamic>` as return type.

Is there any reason for not having that return type here directly? We code generate it into the classes.

If we do it, would it make sense to make the same change for the other `toJson` method?

Reply:
for 'toJsonForProtocol' we can change the return type, but for 'toJson' we cant because 'SerializableEntity' is extended by other types like Enum, and when an enum overrides 'toJson' it returns 'int' or 'String'.
So this change can only be done for 'toJsonForProtocol'.